StyleGuide.md

```diff
@@ -0,0 +1,372 @@
+%% Good
+FileName
+_File
+StoredID
+StoredId
+```
+
+* Use `snake_case` for functions, attributes, and atoms
+```erl
+%% Bad
+'Error'
+badReturn
+read_File
+```
+```erl
+%% Good
+error
+bad_return
+read_file
+```
+
+* Use `SCREAMING_SNAKE_CASE` for macro definitions
+```erl
+%% Bad
+-define(some_value, 100).
+-define(someMacro(), hidden:call()).
+-define(AnotherMacro, "another macro").
+```
+```erl
+%% Good
+-define(SOME_VALUE, 100).
+-define(SOME_MACRO(), hidden:call()).
+-define(ANOTHER_MACRO, "another macro").
+```
+
+* Omit `()` in macro names only if they represent constants
+```erl
+%% Bad
+-define(NOT_CONSTANT, application:get_env(myapp, key)).
+-define(CONSTANT(), 100).
+```
+```erl
+%% Good
+-define(NOT_CONSTANT(), application:get_env(myapp, key)).
+-define(CONSTANT, 100).
+```
+
+* Start names of predicate functions (functions that return a boolean)
+  with `is_` or `has_` prefix
+```erl
+%% Bad
+evenp(Num) -> ...
+```
+```erl
+%% Good
+is_even(Num) -> ...
+```
+
+* Avoid using one-letter variable names and `do_` prefixes for functions,
+  prefer descriptive names
+```erl
+%% Bad
+validate([X | Xs]) ->
+    do_validate(X),
+    validate(Xs).
+```
+```erl
+%% Good
+validate_all([Key | Keys]) ->
+    validate(Key),
+    validate_all(Keys).
+```
+
+* Use `snake_case` for naming applications and modules
+```
+%% Good
+myapp/src/task_server.erl
+```
+
+### Booleans
+
+* avoid `and` and `or` operators, prefer` andalso` and `orelse`, or
+  in guards, where possible, combine expressions with `,` and `;`
+```erl
+%% Bad
+is_atom(Name) and Name =/= undefined.
+is_binary(Task) or is_atom(Task).
+```
+
+```erl
+%% Good
+is_atom(Name) andalso Name =/= undefined.
+is_binary(Task) orelse is_atom(Task).
+```
+
+```erl
+%% Also good
+when is_atom(Name), Name =/= undefined.
+when is_binary(Task); is_atom(Task).
+```
+
+## Formatting
+
+The erlfmt formatter enforces the following rules automatically.
+
+### Whitespace
+
+* avoid trailing spaces
+
+* end each file with a newline
+
+* use single space after comma
+```erl unformatted comma-space
+%% Bad
+application:get_env(kernel,start_pg2)
+```
+```erl formatted comma-space
+%% Good
+application:get_env(kernel, start_pg2)
+```
+
+* use single space before and after binary operator (for example:  `=` , `-`, `|`, `||`, `!`, `->`, or `=>`)
+```erl unformatted binary-space
+%% Bad
+Values=[
+    #{key=>value},
+    #rec{key=value,key2=value2},
+    120-90,
+    Server!Message
+]
+```
+```erl formatted binary-space
+%% Good
+Values = [
+    #{key => value},
+    #rec{key = value, key2 = value2},
+    120 - 90,
+    Server ! Message
+]
+```
+
+* do not use space after symbolic unary operators (for example: `+`, `-`)
+```erl unformatted unary-space
+%% Bad
+Angle = - 45.
+WithParens = - ( + 45 ).
+```
+```erl formatted unary-space
+%% Good
+Angle = -45.
+WithParens = -(+45).
+```
+
+* do not put spaces before or after `( )`, `{ }`, or `[]`  for function/record/maps/lists (declarations, definitions, calls)
```

Comment:
```suggestion
* do not put spaces before or after `( )`, `{ }`, or `[]`  for function/record/maps/lists (declarations, definitions, calls)
```erlang
%% Bad
sort( [ 1,2,3 ] ).
%% Good
sort([1, 2, 3]).
```

And more backticks

Reply:
I just think it is a big sentence with a bit too small example, compared to others